Content.Server/Nyanotrasen/StationEvents/Events/GlimmerShadeSpawnRule.cs

```diff
@@ -0,0 +1,57 @@
+using System.Linq;
+using Robust.Shared.Random;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Shared.Atmos.Miasma;
+using Content.Server.StationEvents.Components;
+using Content.Shared.Psionics.Glimmer;
+
+namespace Content.Server.StationEvents.Events;
+
+internal sealed class GlimmerShadeSpawnRule : StationEventSystem<GlimmerShadeRuleComponent>
+{
+    [Dependency] private readonly IRobustRandom _robustRandom = default!;
+    [Dependency] private readonly GlimmerSystem _glimmerSystem = default!;
+
+    private static readonly string ShadePrototype = "MobShade";
+
+    protected override void Started(EntityUid uid, GlimmerShadeRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        base.Started(uid, component, gameRule, args);
+
+        var rottingPlaces = EntityManager.EntityQuery<RottingComponent, TransformComponent>().ToList();
+        var normalSpawnLocations = EntityManager.EntityQuery<VentCritterSpawnLocationComponent, TransformComponent>().ToList();
+        var hiddenSpawnLocations = EntityManager.EntityQuery<MidRoundAntagSpawnLocationComponent, TransformComponent>().ToList();
```

Comment:
Should probably do a HashSet TransformComponent union of these two, closer to the source of their iteration.

Reply:
it breaks the priority on trying to use the normal ones first and seems like it won't really really simplify things or improve perf

Reply:
or well, let me check again

Reply:
I guess more variance in spawn is good anyway to be less predictable